core/src/main/java/org/apache/james/core/quota/QuotaLimit.java

```diff
@@ -0,0 +1,82 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class QuotaLimit {
+
+    public static QuotaLimit of(QuotaComponent component, QuotaScope scope, String identifier, QuotaType quotaType, Long limit) {
+        return new QuotaLimit(component, scope, identifier, quotaType, limit);
+    }
+
+    private final QuotaComponent quotaComponent;
+    private final QuotaScope quotaScope;
+    private final String identifier;
```

Comment:
you should consistency type class of `identifier` here, 
```suggestion
    private final Username identifier;
```
same with QuotaCurrentValue.java

Reply:
The identifier of QuotaLimit is not necessarily username. It could be other things (domain name for example)